ee/recommendation/core/recommendation_model.py

```diff
@@ -0,0 +1,132 @@
+import mlflow.pyfunc
+
+import numpy as np
+from sklearn import metrics
+from sklearn.svm import SVC
+from sklearn.feature_selection import SequentialFeatureSelector as sfs
+from sklearn.preprocessing import normalize
+from sklearn.decomposition import PCA
+from sklearn.neighbors import KNeighborsClassifier as knc
+
+
+def select_features(X, y):
+    """
+    Dimensional reduction of X using k-nearest neighbors and sequential feature selector.
+    Final dimension set to three features.
+    Params:
+        X: Array which will be reduced in dimension (batch_size, n_features).
+        y: Array of labels (batch_size,).
+    Output: function that reduces dimension of array.
+    """
+    knn = knc(n_neighbors=3)
+    selector = sfs(knn, n_features_to_select=3)
+    selector.fit(X, y)
+
+    def transform(input):
+        return selector.transform(input)
+    return transform
+
+
+def sort_database(X, y):
+    """
+    Random shuffle of training values with its respective labels.
+    Params:
+        X: Array of features.
+        y: Array of labels.
+    Output: Tuple (X_rand_sorted, y_rand_sorted).
+    """
+    import random
+    sort_list = list(range(len(y)))
+    random.shuffle(sort_list)
+    return X[sort_list], y[sort_list]
+
+
+def preprocess(X):
+    """
+    Preprocessing of features (no dimensional reduction) using principal component analysis.
+    Params:
+        X: Array of features.
+    Output: Tuple (processed array of features function that reduces dimension of array).
+    """
+    _, n = X.shape
+    pca = PCA(n_components=n)
+    x = pca.fit_transform(normalize(X))
+
+    def transform(input):
+        return pca.transform(normalize(input))
+
+    return x, transform
+
+
+class SVM_recommendation(mlflow.pyfunc.PythonModel):
+
+    def __init__(self, test=False, **params):
+        f"""{SVC.__doc__}"""
+        if 'probability' not in params.keys():
+            params['probability'] = True
+        assert params['probability'], 'Models need probability parameter set to True'
```

Comment:
if Models needs probability set to True, why not force it for all ? or it could be false ?